### PR TITLE
Table mode cannot be verified by automatic sharding algorithm

### LINKED ISSUE #22926: Enhance the algorithm type check when create sharding table rule
### Which version of ShardingSphere did you use?
master 6f625cf

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
When using the standard way to create sharding table rule, the auto sharding algorithm is not allowed, like `MOD`, `HASH_MOD`.

### Actual behavior
Can now use the auto sharding algorithm to create standard sharding table rule
```sql
mysql> CREATE SHARDING TABLE RULE t_order_item (
    -> DATANODES("ds_${0..1}.t_order_item"),
    -> DATABASE_STRATEGY(TYPE="standard",SHARDING_COLUMN=user_id,SHARDING_ALGORITHM(TYPE(NAME="MOD",PROPERTIES("sharding-count"=4))))
    -> );
Query OK, 0 rows affected (0.38 sec)

mysql> 
```

### Reason analyze (If you can)
Similar to https://github.com/apache/shardingsphere/pull/22900

### PR DESCRIPTION
**Pull request by gxxiong**
Fixes #22926 .

Changes proposed in this pull request:
  -Table mode cannot be verified by automatic sharding algorithm

Before committing this PR, I'm sure that I have checked the following options:
- [X] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [X] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [X] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [X] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/fixture/sharding/HintInlineShardingAlgorithmFixture.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    
+    private Closure<?> createClosure() {
+        Closure<?> result = new InlineExpressionParser(algorithmExpression).evaluateClosure().rehydrate(new Expando(), null, null);
+        result.setResolveStrategy(Closure.DELEGATE_ONLY);
+        return result;
+    }
+    
+    @Override
+    public String getType() {
+        return "HINT_INLINE";
```
Comment (RaigorJiang) [inline features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/fixture/sharding/HintInlineShardingAlgorithmFixture.java:79]: Sorry for being late, why do we need to add `HINT_INLINE`, `INLINE`, `MOD` algorithms, can the built-in algorithms not be used?

Comment (gxxiong) [reply features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/fixture/sharding/HintInlineShardingAlgorithmFixture.java:79]: Need to build shardingAlgorithm to judge whether it is ShardingAutoTableAlgorithm, algorithm type cannot be built

Comment (RaigorJiang) [reply features/sharding/distsql/handler/src/test/java/org/apache/shardingsphere/sharding/distsql/fixture/sharding/HintInlineShardingAlgorithmFixture.java:79]: But these SPIs have the same type name as the built-in algorithms (like `INLINE`), can we tweak this?

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
Hi @gxxiong 
This is an important PR, thanks for your contribution!

**Pull request merged**